packages/flame/lib/src/experimental/geometry/shapes/shape.dart

```diff
@@ -0,0 +1,86 @@
+import 'dart:ui';
+
+import 'package:vector_math/vector_math_64.dart';
+
+import '../../../game/transform2d.dart';
+import 'circle.dart';
+import 'polygon.dart';
+import 'rectangle.dart';
+import 'rounded_rectangle.dart';
+
+/// Base class for various 2D geometric primitives defined on a Cartesian
+/// coordinate plane.
+///
+/// Implementations include:
+///   - [Circle]
+///   - [Polygon]
+///   - [Rectangle]
+///   - [RoundedRectangle]
+abstract class Shape {
+  /// True if the shape is "closed", in the sense that it has an interior. For
+  /// example, a closed shape can be filled with a paint.
+  bool get isClosed => true;
+
+  /// True if the shape is convex, i.e. a line segment connecting any two points
+  /// of the shape would lie completely within the shape.
+  bool get isConvex;
+
+  /// The length of the shape's boundary. For some more complicated shapes this
+  /// can be computed approximately.
+  double get perimeter;
```

Comment:
Curius of what this is used for

Reply:
Say, if you want to move a component along the border of the shape, you may want to know the length of the border in order to calculate the timing.
It may not be very useful, but it doesn't impose any extra cost, and it's a fundamental geometric property of a shape.